dm/ctl/simulator/bw_list.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package simulator
+
+import (
+	"context"
+
+	"github.com/pingcap/dm/dm/config"
+	"github.com/pingcap/dm/dm/ctl/common"
+	"github.com/pingcap/dm/dm/pb"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb-tools/pkg/dbutil"
+	"github.com/spf13/cobra"
+)
+
+type bwListResult struct {
+	Result         bool                `json:"result"`
+	Msg            string              `json:"msg"`
+	DoTables       map[string][]string `json:"do-tables,omitempty"`
+	IgnoreTables   map[string][]string `json:"ignore-tables,omitempty"`
+	WillBeFiltered string              `json:"will-be-filtered,omitempty"`
+}
+
+// NewBWListCmd creates a BWList command
+func NewBWListCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "bw-list [-w worker] [-T table] <config-file>",
+		Short: "check the black-white-list info for tables",
+		Run:   bwListFunc,
+	}
+	tableList = tableList[:0]
+	cmd.Flags().StringSliceVarP(&tableList, "table", "T", []string{}, "the table name we want to check for the table route")
+	return cmd
+}
+
+func bwListFunc(cmd *cobra.Command, _ []string) {
+	content, err := common.GetFileContent(cmd.Flags().Arg(0))
```

Comment:
Can we extract these args parsing as a function since they are the same with `table-route`?

Reply:
I don't understand.. Could you please give an example?